### ISSUE TITLE
Cannot find target `:descriptor_proto_srcs`

### ISSUE THREAD

**Issue by n1c00o** [protocolbuffers/protobuf#10296]
**What version of protobuf and what language are you using?**
Version: v21.3
Language: none

**What operating system (Linux, Windows, ...) and version?**

Arch Linux rolling

**What runtime / compiler are you using (e.g., python version or gcc version)**

GCC (I think) with Bazel

**What did you do?**
1. Define the repository in Bazel

```bazel
http_archive(
    name = "com_google_protobuf",
    sha256 = "1e958b62debbb46ebefa16d848741d17c88dc018dd57b729c0cd58965380f3f8",
    strip_prefix = "protobuf-21.3",
    urls = [
        "https://github.com/protocolbuffers/protobuf/archive/refs/tags/v21.3.tar.gz",
    ],
)
```

2. Defines a target which requires `@com_google_protobuf//:descriptor_proto_srcs` in data
3. Try to build the target
4. Get a Bazel error

```bash
(omitted...) no such target '@com_google_protobuf//:descriptor_proto_srcs': target 'descriptor_proto_srcs' not declared in package '' (omitted...)
```

**What did you expect to see**

Nothing related to this

**What did you see instead?**

```bash
(omitted...) no such target '@com_google_protobuf//:descriptor_proto_srcs': target 'descriptor_proto_srcs' not declared in package '' (omitted...)
```

The BUILD.bazel file distributed in the archive does not provide a target called `:descriptor_proto_srcs` nor provide a public target including the source file `src/google/protobuf/descriptor.proto`.

The only way to get the file is by using the deprecated filegroup `:well_known_protos` but the target is marked as deprecated and the comment notices us to use `:descriptor_proto_srcs` for the source file descriptor.proto.

It would be helpful to either make the filegroup `:built_in_runtime_protos` public or add `src/google/protobuf/descriptor.proto` to a publicly available filegroup and updating the comment in the deprecation notice of `:well_known_protos`.

The deprecated `well_known_protos` target definition:  *(BUILD.bazel line 159 in the distributed archive)*
```bazel

# DEPRECATED: Prefer :well_known_type_protos for the Well-Known Types
# (https://developers.google.com/protocol-buffers/docs/reference/google.protobuf)
# or :descriptor_proto(_srcs) for descriptor.proto (source), or
# :compiler_plugin_proto for compiler/plugin.proto.
filegroup(
    name = "well_known_protos",
    srcs = [
        "src/google/protobuf/compiler/plugin.proto",
        "src/google/protobuf/descriptor.proto",
        ":well_known_type_protos",
    ],
    deprecation = "Prefer :well_known_type_protos instead.",
    visibility = ["//visibility:public"],
)
```

The new `well_known_type_protos` target definition:  *(BUILD.bazel line 174 in the distributed archive)*
```bazel
filegroup(
    name = "well_known_type_protos",
    srcs = [
        "src/google/protobuf/any.proto",
        "src/google/protobuf/api.proto",
        "src/google/protobuf/duration.proto",
        "src/google/protobuf/empty.proto",
        "src/google/protobuf/field_mask.proto",
        "src/google/protobuf/source_context.proto",
        "src/google/protobuf/struct.proto",
        "src/google/protobuf/timestamp.proto",
        "src/google/protobuf/type.proto",
        "src/google/protobuf/wrappers.proto",
    ],
    visibility = ["//visibility:public"],
)
```

The `built_in_runtime_protos` target definition which includes the required file but is not public:  *(BUILD.bazel line 191 in the distributed archive)*
```bazel
filegroup(
    name = "built_in_runtime_protos",
    srcs = [
        "src/google/protobuf/compiler/plugin.proto",
        "src/google/protobuf/descriptor.proto",
    ],
    visibility = ["//:__subpackages__"],
)
```

More stuff can be found by downloading the archive at <https://github.com/protocolbuffers/protobuf/archive/refs/tags/v21.3.tar.gz> (same goes for the protobuf-all version)

**Comment by n1c00o** [comment]
As for now, the solution is to use the deprecated `:well_known_protos` target

**Comment by n1c00o** [comment]
Side note: it seems that the current main branch does include an alias for `:descriptor_proto_srs` [here](https://github.com/protocolbuffers/protobuf/blob/main/BUILD.bazel#L113)

**Comment by n1c00o** [comment]
Okay so it looks like my issue was fixed in recent commits that are not included in the v21.3 release, sorry for the inconvenience

**Comment by zhangskz** [comment]
Looks like the alias for `:descriptor_proto_srcs` in main is not part of the 21.3 release. Closing since there is the `:descriptor_proto` proto_library and `:well_known_types` for 21.3 and the next 22 release should add the new `:descriptor_proto_srcs` alias. This likely doesn't warrant another patch release for 21.

Please feel free to reopen if this isn't sufficient!
